s3-source-connector/README.md

```diff
@@ -21,7 +21,290 @@ published into the corresponding Kafka topic.
+
+  Configuration: ```input.format=avro```.
+
+- Complex structure, where file is in Apache Parquet file format.
+
+  Configuration: ```input.format=parquet```.
+-
+- Complex structure, where file is in bytes format.
+
+  Configuration: ```input.format=bytes```.
+
+The connector can output the following fields from records into the
+output: the key, the value, the timestamp, the offset and headers. (The set and the order of
+output: the key, the value, the timestamp, the offset and headers. The set of
+these output fields is configurable.) The field values are separated by comma.
+
+#### JSONL Format example
+
+For example, if we output `key,value,offset,timestamp`, a record line might look like:
+
+```json
+ { "key": "k1", "value": "v0", "offset": 1232155, "timestamp":"2020-01-01T00:00:01Z" }
+```
+
+org.apache.kafka.connect.json.JsonConverter is used internally to convert this data and make output files human-readable.
+
+**NB!**
+
+- Value/Key schema will not be presented in output kafka event, even if `value.converter.schemas.enable` property is `true`,
+  however, if this is set to true, it has no impact at the moment.
+
+#### Parquet or Avro format example
+
+For example, if we input `key,offset,timestamp,headers,value`, an input - Parquet schema in an s3 object might look like this:
+```json
+{
+    "type": "record", "fields": [
+      {"name": "key", "type": "RecordKeySchema"},
+      {"name": "offset", "type": "long"},
+      {"name": "timestamp", "type": "long"},
+      {"name": "headers", "type": "map"},
+      {"name": "value", "type": "RecordValueSchema"}
+  ]
+}
+```
+where `RecordKeySchema` - a key schema and `RecordValueSchema` - a record value schema.
+This means that in case you have the record and key schema like:
+
+Key schema:
+```json
+{
+  "type": "string"
+}
+```
+
+Record schema:
+```json
+{
+    "type": "record", "fields": [
+      {"name": "foo", "type": "string"},
+      {"name": "bar", "type": "long"}
+  ]
+}
+```
+the final `Avro` schema for `Parquet` is:
+```json
+{
+    "type": "record", "fields": [
+      {"name": "key", "type": "string"},
+      {"name": "offset", "type": "long"},
+      {"name": "timestamp", "type": "long"},
+      {"name": "headers", "type": "map", "values": "long"},
+      { "name": "value",
+        "type": "record",
+        "fields": [
+          {"name": "foo", "type": "string"},
+          {"name": "bar", "type": "long"}
+        ]
+      }
+  ]
+}
+```
+**NB!**
+
+- Connector works just fine with and without Schema Registry
+
+## Usage
+
+### Connector Configuration
+
+> **Important Note** Since this connector is developed aligning it with S3 sink connector,
+> and since version `2.6`, all existing configuration in S3 sink
+is deprecated and will be replaced with new one during a certain transition period (within 2-3 releases). Most of the
+> configuration parameters remain same.
+
+List of deprecated configuration parameters:
```

Comment:
Lets remove the deprecated config params and not support them in the S3 Source Connector so we dont have any legacy we need to clean up in the future.